src/tests/test_finite_difference.py

```diff
@@ -1,64 +1,60 @@
 from __future__ import absolute_import
 from builtins import range
-from __init__ import ForceBalanceTestCase
-import unittest
 import numpy
 import forcebalance
 import re, sys
 from math import cos, sin, pi
+import pytest
+# class TestFiniteDifference(ForceBalanceTestCase):
+# class TestFiniteDifference:
+#     # def setUp(self):
+#     @classmethod
+# def setup_class(cls):
+    # functions is a list of 3-tuples containing (function p, d/dp, d^2/dp^2)
+functions = []
 
-class TestFiniteDifference(ForceBalanceTestCase):
-    def setUp(self):
```

Comment:
Do we gain anything by having `setUp` run in a class, rather than defining important variables for our tests at the module level? Is there anything bad about defining these at the module level?

Reply:
We should use `setup_method` and `teardown_method` wherever possible.

Reply:
The possible drawback of defining setup and teardown at the module level is if we want to run the following sequence of tests in a single module (which happens most often in test_engine.py):

setupA-testA1-testA2-teardownA (e.g. test optimized geometries and vibrational frequencies for Gromacs vs. AMBER for the alanine-glutamate system)
setupB-testB1-testB2-teardownB (e.g. test optimized geometries and vibrational frequencies for Gromacs vs. AMBER for the water cluster system)